### PR TITLE
Osmosis v3 Semver updates

### PR DESCRIPTION
**Pull request by faddat**
Closes: #788 

## Description
This PR fixes issues found in semver for osmosis v3

it also:

* Updates tendermint to v0.34.15
* Provides rocksdb support
* Updates tm-db

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by mattverse** [comment]
Planning to get this reviewed / merged post-superfluid

**Comment by faddat** [comment]
This no longer uses the replace statement for gorocksdb, and instead uses tm-db v0.6.7

**Review by mattverse — COMMENTED** [review]
Did you intentionally update the go version and the cosmos-sdk versions as well?

**Comment by faddat** [comment]
no and thank you matt!

I remember rememebering that that thad happened, I will make it reverted....

**Comment by mattverse** [comment]
Nice! tysm!!

File: go.mod [hunk 1]
```diff
@@ -1,33 +1,122 @@
 module github.com/osmosis-labs/osmosis/v3
 
-go 1.15
+go 1.17
```
Comment (alexanderbez) [inline go.mod:3]: Why not just bump to 1.18 (including our CI plumbing) while we're at it?

File: go.mod [hunk 2]
```diff
@@ -1,33 +1,122 @@
 module github.com/osmosis-labs/osmosis/v3
```
Comment (alexanderbez) [inline go.mod:1]: Sorry, silly question. Being that we're on v7 of osmosis, Osmosis should technically be on v7 in the repo too, no?

Comment (ValarDragon) [reply go.mod:1]: this is a PR to the v4 branch

Comment (alexanderbez) [reply go.mod:1]: derppp sorry :p

File: go.mod (hunk 1 above)
Comment (faddat) [reply go.mod:3]: Absolutely

**Comment by faddat** [comment]
@alexanderbez I’m going to handle the go 1.18 stuff, and then open a new pr to bring the iavl changes into the past work

**Pull request merged**